### ISSUE TITLE
Missing `NgFor` directive not reported

### ISSUE THREAD

**Issue by zoechi** [dart-lang/angular_analyzer_plugin#175]
```html
<mdl-navigation-link *ngFor="let item of domainResources"
                     [routerLink]="['/Resource', 'Search', {'resourceType': item.resourceName}]"
                     class="mdl-color-text--blue-grey-400 ">{{item.resourceName}}
</mdl-navigation-link>
```

causes
```
ERROR: Missing type arguments for map literal. ([fhir_client] lib/webui/resource_overview/resource_overview_component.html:3)
ERROR: Undefined name 'item'. ([fhir_client] lib/webui/resource_overview/resource_overview_component.html:3)
ERROR: Undefined name 'item'. ([fhir_client] lib/webui/resource_overview/resource_overview_component.html:4)
```
but no error for an unknown `NgFor` but building causes

```
Build error:
Transform TemplateCompiler on fhir_client|lib/webui/resource_overview/resource_overview_component.ng_meta.json threw error: Template parse errors:
line 2, column 1 of ResourceOverviewComponent: ParseErrorLevel.FATAL: Property binding ngForOf not used by any directive on an embedded template
<mdl-navigation-link *ngFor="let item of domainResources"
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
and the directive is actually missing because I removed it from the Angular2 transformer configuration and haven't added it to this component yet because the analyser didn't point it out.

**Comment by zoechi** [comment]
DartAnalyzer (in WebStorm) stopped for some reason and was restarted automatically by webstorm, then these missing `NgFor`s were reported.

**Comment by MichaelRFairhurst** [comment]
The plugin does sometimes crash. Do you have any crash logs anywhere? We will also fuzz test it before too long I hope, but that will only reveal so many scenarios.

**Comment by zoechi** [comment]
I tried to find something but there wasn't anything in the logs that looked useful. I'll keep looking.
